openbook_posts/views/post_comments/serializer_fields.py

```diff
@@ -0,0 +1,44 @@
+from rest_framework.fields import Field
+
+
+class RepliesCountField(Field):
+    def __init__(self, **kwargs):
+        kwargs['source'] = '*'
+        kwargs['read_only'] = True
+        super(RepliesCountField, self).__init__(**kwargs)
+
+    def to_representation(self, post_comment):
+        request = self.context.get('request')
+        request_user = request.user
+
+        replies_count = request_user.get_replies_count_for_post_comment(post_comment=post_comment)
+
+        return replies_count
+
+
+class RepliesField(Field):
+    SORT_CHOICE_TO_QUERY = {
+        'DESC': '-created',
+        'ASC': 'created'
+    }
+    DEFAULT_REPLY_COUNT = 2
+
+    def __init__(self, post_comment_reply_serializer=None, **kwargs):
+        kwargs['source'] = '*'
+        kwargs['read_only'] = True
+        self.post_comment_reply_serializer = post_comment_reply_serializer
+        super(RepliesField, self).__init__(**kwargs)
+
+    def to_representation(self, post_comment):
+        request = self.context.get('request')
+        request_user = request.user
+        request_data = request.query_params.dict()
+        sort = request_data.get('sort', 'DESC')
```

Comment:
Dangerous. Never use data from requests that does not go through a serializer beforehand.

Sanitize the available options in a serializer in the view, and pass the chosen one to this fiedl as a `replies_sort` argument as in 

```response_data = TheOuterSerializer(post_comment_replies, many=True, context={"request": request, "replies_sort": sanitized_replies_sort}).data```

Please check you're not accessing raw request data anywhere else.

Reply:
If the user can choose between `ASC` or `DESC` I recommend using a [choicefield](https://www.django-rest-framework.org/api-guide/fields/#choicefield)! Really useful.

Reply:
yea agreed, i had this concern too (im not doing this anywhere else dont worry:) ) but this field is a response serialIzier field, so i thought we already check if this is a valid entry - during the view... if its not .. we would never reach this code. 

If we do reach the code, the sort was one of the valid choices we allow in the request serializer. So i chose not to re-check this again here. 

I can do it though :)

Reply:
Do it please. People using this Field in the future might not be aware that this field is dependent on a query parameter named. Pass it explicitly on the request context and fail if not given. 

In general, we should avoid using any request data other than the `request.user` in the serializer fields. Those can be passed explicitly as part of the context as shown above. The user is passed just like that, but within the `request` obj.

Reply:
> If the user can choose between `ASC` or `DESC` I recommend using a [choicefield](https://www.django-rest-framework.org/api-guide/fields/#choicefield)! Really useful.

Yep @evict this is already a choice field, joel wants to double check to avoid assumptions of sanitisation in the future use cases.